library/Fission/IPFS/DNSLink/Class.hs

```diff
@@ -0,0 +1,11 @@
+module Fission.IPFS.DNSLink.Class (MonadDNSLink (..)) where
+
+import           Network.IPFS.CID.Types
+import           Servant
+
+import           Fission.AWS.Route53.Class
+import           Fission.AWS.Types as AWS
+import           Fission.Prelude
+
+class MonadRoute53 m => MonadDNSLink m where
+  set :: Maybe AWS.Subdomain -> CID -> m (Either ServerError AWS.DomainName)
```

Comment:
@bechurch To contain the `class`.

I'm guessing that you're wondering why this file since there's also a `Fission.IPFS.DNSLink`. If the upper-level (which reexports this module) later contains a bunch of helper functions, we may want to import _just_ the `class` to avoid module cycles elsewhere. This is the same reason as the `*.Types` modules.